### ISSUE TITLE
Better support for transferred starred repos

### ISSUE THREAD

**Issue by szpak** [vfeskov/gitpunch#88]
I really like GitPunch which allows me to follow favorite projects (which was the killer feature especially in the times it was not supported by GitHub)!

I see one field for improvements. When a starred repo is transferred, e.g. from an [user account](https://github.com/aalmiray/kordamp-gradle-plugins) to an [organization](https://github.com/kordamp/kordamp-gradle-plugins/) there are 2 notifications sent:
![image](https://user-images.githubusercontent.com/148013/71645809-25659e80-2cde-11ea-9a3a-3bd9daa765cb.png)

It's not a big deal, I could unsubscribe manually, but maybe it could be somehow filtered.

**Comment by vfeskov** [comment]
yeah the root cause is that i'm using repo names as IDs instead of repo's actual numeric IDs. Refactoring to use numeric IDs is a big project since releases are requested by repo name and not it's ID, but it's not impossible

**Comment by szpak** [comment]
It's not a big problem. It's probably not worth to do the big refactoring.

**Comment by szpak** [comment]
Btw, maybe it would be easier (and possible) to just detect duplicates in the project stream and filter out the "older one"?

**Comment by bersbersbers** [comment]
> I could unsubscribe manually

How do you do that? Can you do that from GitHub somehow (that's where I'm struggling), or do you have to go to pitpunch.com?

**Comment by AlexWayfer** [comment]
> > I could unsubscribe manually
> 
> How do you do that? Can you do that from GitHub somehow (that's where I'm struggling), or do you have to go to gitpunch.com?

There is just a bell button on the GitPunch.com to mute repos:

![image](https://user-images.githubusercontent.com/6510020/107217672-8ef6c700-6a1f-11eb-991f-79cfd8b3e850.png)

**Comment by bersbersbers** [comment]
@AlexWayfer yes, I just became aware of that (hadn't visited gitpunch.com in a looong time, because all UI I need on a daily basis is on github.com). I just asked myself how to unsubscribe from a renamed repo on github.com (you probably cannot).

**Comment by bersbersbers** [comment]
> Btw, maybe it would be easier (and possible) to just detect duplicates in the project stream and filter out the "older one"?

I guess so. I had starred https://github.com/rg3/youtube-dl once, which is now https://github.com/ytdl-org/youtube-dl - so I was getting duplicate announcements.

But a simple GET request can give you the new URL:

```
C:\Users\bers>curl https://github.com/rg3/youtube-dl
<html><body>You are being <a href="https://github.com/ytdl-org/youtube-dl">redirected</a>.</body></html>
```

And you don't even need to download the body:

```
C:\Users\bers>curl --head https://github.com/rg3/youtube-dl 2>&1 | grep Location
Location: https://github.com/ytdl-org/youtube-dl
```

This can be done one a day/week/month for housekeeping, or maybe implemented on-the-fly to avoid duplicates (although the latter would probably massively increase the number of requests).
